test/scripts/e2e_subs/rekey.sh

```diff
@@ -54,35 +52,57 @@ fi
 
-${gcmd} clerk send -a 1 -f ${ACCOUNTD} -t ${ACCOUNTD} --rekey-to ${ACCOUNT} -o ${TEMPDIR}/txn2.tx
-${gcmd} clerk send -a 1 --from-program "${TEMPDIR}/simple.teal" -t ${ACCOUNTD} -o ${TEMPDIR}/txn3.tx
-cat ${TEMPDIR}/txn2.tx ${TEMPDIR}/txn3.tx > ${TEMPDIR}/group1.tx
+${gcmd} clerk send -a 1 -f "${ACCOUNTD}" -t "${ACCOUNTD}" --rekey-to "${ACCOUNT}" -o "${TEMPDIR}/txn2.tx"
+${gcmd} clerk send -a 1 --from-program "${TEMPDIR}/simple.teal" -t "${ACCOUNTD}" -o "${TEMPDIR}/txn3.tx"
+cat "${TEMPDIR}/txn2.tx" "${TEMPDIR}/txn3.tx" > "${TEMPDIR}/group1.tx"
 
 # Build + sign group
-${gcmd} clerk group -i ${TEMPDIR}/group1.tx -o ${TEMPDIR}/group1_grouped.tx
-${gcmd} clerk split -i ${TEMPDIR}/group1_grouped.tx -o ${TEMPDIR}/group1_split.txn
-${gcmd} clerk sign -i ${TEMPDIR}/group1_split-0.txn -o ${TEMPDIR}/group1_split-0.stxn
-cat ${TEMPDIR}/group1_split-0.stxn ${TEMPDIR}/group1_split-1.txn > ${TEMPDIR}/group1_signed.stxn
+${gcmd} clerk group -i "${TEMPDIR}/group1.tx" -o "${TEMPDIR}/group1_grouped.tx"
+${gcmd} clerk split -i "${TEMPDIR}/group1_grouped.tx" -o "${TEMPDIR}/group1_split.txn"
+${gcmd} clerk sign -i "${TEMPDIR}/group1_split-0.txn" -o "${TEMPDIR}/group1_split-0.stxn"
+cat "${TEMPDIR}/group1_split-0.stxn" "${TEMPDIR}/group1_split-1.txn" > "${TEMPDIR}/group1_signed.stxn"
 
 # Broadcast group (should succeed)
-${gcmd} clerk rawsend -f ${TEMPDIR}/group1_signed.stxn
+${gcmd} clerk rawsend -f "${TEMPDIR}/group1_signed.stxn"
 
 # Regular rekeying test
-algokey generate > ${TEMPDIR}/rekey
-mnemonic=$(grep 'Private key mnemonic:' < ${TEMPDIR}/rekey | sed 's/Private key mnemonic: //')
-ACCOUNTC=$(grep 'Public key:' < ${TEMPDIR}/rekey | sed 's/Public key: //')
+algokey generate > "${TEMPDIR}/rekey"
+mnemonic=$(grep 'Private key mnemonic:' < "${TEMPDIR}/rekey" | sed 's/Private key mnemonic: //')
+ACCOUNTC=$(grep 'Public key:' < "${TEMPDIR}/rekey" | sed 's/Public key: //')
 ${gcmd} account import -m "${mnemonic}"
 
-${gcmd} clerk send -a 10000000 -f ${ACCOUNT} -t ${ACCOUNTB} --rekey-to ${ACCOUNTC}
+${gcmd} clerk send -a 10000000 -f "${ACCOUNT}" -t "${ACCOUNTB}" --rekey-to "${ACCOUNTC}"
 
-${gcmd} clerk send -a 13000000 -f ${ACCOUNT} -t ${ACCOUNTB} -o ${TEMPDIR}/ntxn
-${gcmd} clerk sign -S ${ACCOUNTC} -i ${TEMPDIR}/ntxn -o ${TEMPDIR}/nstxn
-${gcmd} clerk rawsend -f ${TEMPDIR}/nstxn
+${gcmd} clerk send -a 13000000 -f "${ACCOUNT}" -t "${ACCOUNTB}" -o "${TEMPDIR}/ntxn"
+${gcmd} clerk sign -S "${ACCOUNTC}" -i "${TEMPDIR}/ntxn" -o "${TEMPDIR}/nstxn"
+${gcmd} clerk rawsend -f "${TEMPDIR}/nstxn"
 
-BALANCEB=$(${gcmd} account balance -a ${ACCOUNTB} | awk '{ print $1 }')
-if [ $BALANCEB -ne 23000000 ]; then
-    date '+e2e_subs/rekey.sh FAIL wanted balance=23000000 but got ${BALANCEB} %Y%m%d_%H%M%S'
+BALANCEB=$(${gcmd} account balance -a "${ACCOUNTB}" | awk '{ print $1 }')
+if [ "$BALANCEB" -ne 23000000 ]; then
+    date "+e2e_subs/rekey.sh FAIL wanted balance=23000000 but got ${BALANCEB} %Y%m%d_%H%M%S"
+    false
+fi
+
+# Rekey from A to C back to A [A -> C -> A].
```

Comment:
this looks different from the ticket requirements 

> add a test which rekeys from A to B back to A [A -> B -> A] and another test which rekeys from A to B to C [A -> B -> C]

Reply:
But what does that add? Ben had that written but I advised against it (I hadn't seen the ticket) because it felt unnecessarily redundant to me. Once a re-key is tested to work with the new key and not work with the previous key, what does doing it again add?

Reply:
I'm happy to add new tests if you think it's worth it.  Please let me know soon, though, as it'd be great to get this merged before end of day.  Thanks!

Reply:
we have basic tests A->B and back already but not to C. I'm not familiar with the implementation well enough. If it's redundant I'm fine with it.

Reply:
Thanks for responding @algorandskiy , I'll mark this conversation as resolved.